design/pub-sub.md

```diff
@@ -0,0 +1,42 @@
+---
+title: pub-sub
+target-version: 0.1.0
+---
+
+# PubSub
+
+## Summary
+Yorkie needs to share events happening in documents between different clients. For this, we implement this feature using gRPC server-side stream and the PubSub pattern.
+
+### Goals
+We should be able to share events with other clients who are subscribing to Documents.
+
+## Proposal Details
+
+### How does it work?
+Yorkie implements WatchDocuments API using [gRPC server-side stream](https://grpc.io/docs/languages/go/basics/#server-side-streaming-rpc) to deliver the events that have occurred to other clients.
+
+```protobuf
+// api/yorkie.proto
+
+service Yorkie {
+    ...
+    rpc WatchDocuments (WatchDocumentsRequest) returns (stream WatchDocumentsResponse) {}
+}
+```
+
+And to manage the event delivery target, we are using the [PubSub pattern](https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern). 
+We can learn more by looking at the [sync package](https://github.com/yorkie-team/yorkie/tree/main/yorkie/backend/sync) we are implementing.
+ 
+The order of operation is as follows.
+
+1. The WatchDocuments API creates a `Subscription` instance with the client and document keys.
+2. Subscription instance internally manages the `DocEvent channel`, and the WatchDocuments API uses a select statement to retrieve events passed to the Subscription instance.
+3. The client can deliver the event to the Subscription instances that are subscribing to the same document through the `PubSub.Publish(publisherID *time.ActorID, topic string, event DocEvent)` method.
```

Comment:
3. The client can deliver the event to the Subscription instances that are subscribed to the same document through the `PubSub.Publish(publisherID *time.ActorID, topic string, event DocEvent) method.`